src/main/java/ch/njol/skript/effects/EffBuildRestrictions.java

```diff
@@ -0,0 +1,97 @@
+package ch.njol.skript.effects;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.aliases.ItemType;
+import ch.njol.skript.doc.*;
+import ch.njol.skript.lang.Effect;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import com.destroystokyo.paper.Namespaced;
+import org.bukkit.event.Event;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.ItemMeta;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+@Name("Apply Build Restrictions")
+@Description("Allow or prevent an item to destroy or be placed on certain types of blocks while in /gamemode adventure.")
+@Examples("allow player's tool to destroy (stone and oak wood planks)")
+@Since("INSERT VERSION")
+@RequiredPlugins("Paper")
+public class EffBuildRestrictions extends Effect {
+
+	static {
+		if (Skript.methodExists(ItemMeta.class, "setDestroyableKeys")) {
+			Skript.registerEffect(EffBuildRestrictions.class,
+				"allow %~itemtypes% to (:(destroy|break|mine|be placed on)) %itemtypes%",
+				"(disallow|prevent) %~itemtypes% from (:(destroying|breaking|mining|being placed on)) %itemtypes%");
+			// should this require notnull? and can the patterns here be made any better?
+		}
+	}
+
+	private boolean allow;
+	private boolean destroy;
+	@SuppressWarnings("NotNullFieldNotInitialized")
+	private Expression<ItemType> items;
+	@SuppressWarnings("NotNullFieldNotInitialized")
+	private Expression<ItemType> modifyWith;
```

Comment:
Should be named something like keys, deltaKeys, keysToChange, etc. 
optional though, not a required change